chainer/functions/tensor_network.py

```diff
@@ -0,0 +1,264 @@
+import math
+
+import numpy
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+def _as_vec(x):
+    return x.reshape(x.size)
+
+
+def _as_mat(x):
+    return x.reshape(x.shape[0], x.size // x.shape[0])
+
+
+def _empty_like(x):
+    if isinstance(x, cuda.GPUArray):
+        return cuda.empty_like(x)
+    else:
+        return numpy.empty_like(x)
+
+
+class TensorNetwork(function.Function):
+
+    def __init__(self, in_shape, out_size, nobias=False,
+                 initialW=None, initial_bias=None):
+
+        self.W = None
+        self.gW = None
+        self.V1 = None
+        self.gV1 = None
+        self.V2 = None
+        self.gV2 = None
+        self.b = None
+        self.gb = None
+
+        assert len(in_shape) == 2
```

Comment:
Yes. `in_shape` is a pair of ints. First (resp. Second) integer represents the number of features of first (rest. second) input tensor.
I come to think that we should rename the name `in_shape` to `in_sizes`.